blocklist-client/src/client/risk_client.rs

```diff
@@ -0,0 +1,352 @@
+use crate::common::error::Error;
+use crate::common::{BlocklistStatus, RiskSeverity};
+use crate::config::RiskAnalysisConfig;
+use reqwest::{Client, Response, StatusCode};
+use serde::Deserialize;
+use tracing::debug;
+const API_BASE_PATH: &str = "/api/risk/v2/entities";
+
+#[derive(Deserialize, Debug)]
+struct RegistrationResponse {
+    address: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct RiskResponse {
+    risk: Option<String>,
+    #[serde(rename = "riskReason")]
+    risk_reason: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct RiskAssessment {
+    pub severity: RiskSeverity,
+    pub reason: String,
+}
+
+/// Register the user address with provider to run subsequent risk checks
+async fn register_address(
+    client: &Client,
+    config: &RiskAnalysisConfig,
+    address: &str,
+) -> Result<RegistrationResponse, Error> {
+    let api_url = register_address_path(&config.api_url);
+    let body = serde_json::json!({ "address": address });
+
+    debug!("Beginning registration for address: {address}");
+
+    let response = client
+        .post(&api_url)
+        .header("Token", &config.api_key)
+        .header("Content-Type", "application/json")
+        .json(&body)
+        .send()
+        .await?;
+
+    let checked_response = check_api_response(response).await?;
+    checked_response
+        .json::<RegistrationResponse>()
+        .await
+        .map_err(Error::from)
+}
+
+/// Check risk status associated with a registered address
+async fn get_risk_assessment(
+    client: &Client,
+    config: &RiskAnalysisConfig,
+    address: &str,
+) -> Result<RiskAssessment, Error> {
+    let api_url = risk_assessment_path(&config.api_url, address);
+    debug!("Beginning risk assessment for address: {address}");
+
+    let response = client
+        .get(&api_url)
+        .header("Token", &config.api_key)
+        .send()
+        .await?;
+
+    let checked_response = check_api_response(response).await?;
+    let resp = checked_response.json::<RiskResponse>().await?;
+
+    match resp.risk {
+        Some(risk_str) => {
+            let severity = match risk_str.as_str() {
+                "Low" => RiskSeverity::Low,
+                "Medium" => RiskSeverity::Medium,
+                "High" => RiskSeverity::High,
+                "Severe" => RiskSeverity::Severe,
+                _ => return Err(Error::InvalidRiskValue(risk_str)),
+            };
```

Comment:
Nit: It would be more idiomatic to put this match statement into a `FromStr` implementation for `RiskSeverity`. With that, this code would simplify to:
```suggestion
            let severity = risk_str.parse().map_err(|_| Error::InvalidRiskValue(risk_str))?;
```

The `.map_err` could potentially be skipped if the error type used in the `FromStr` implementation implements `Into<Error>` for the common error type.

Reply:
Note that this is not needed if we deserialize the response as a `RiskAssessment` directly as per my previous comment.